### PR TITLE
fix: add `--no-pager` option to git command

### PR DESCRIPTION
**Pull request by CaptainOfPhB**
#### Description (required)

The previous git command can not work as expected. With no `--no-pager` option, git command will output the paging result, so `execSync` can't get the returned result.

image

After adding the `--no-pager` option(meanwhile set `pretty` to `%cI`and wrap `filepath` with double quote), we get what we want.

image

#### Related issues & labels (optional)

- Suggested label: update doc

### CODE REVIEW

**Review by TheOtterlord — COMMENTED** [review]
Thanks for the PR @CaptainOfPhB!

I see what you mean about the pager when using the command in the terminal. However, I found that calling `execSync` with the git command works without `--no-pager`, and only in the terminal did I have that issue.

Looking at the new formatting, running `execSync` with `--pretty=%cI`, it threw an invalid date error, displaying the last character as a `\n`, something that doesn't happen with the old argument. Did you have a specific issue with the old formatting?

**Comment by CaptainOfPhB** [comment]
> Thanks for the PR @CaptainOfPhB!
> 
> I see what you mean about the pager when using the command in the terminal. However, I found that calling `execSync` with the git command works without `--no-pager`, and only in the terminal did I have that issue.
> 
> Looking at the new formatting, running `execSync` with `--pretty=%cI`, it threw an invalid date error, displaying the last character as a `\n`, something that doesn't happen with the old argument. Did you have a specific issue with the old formatting?

Thank you for your reply!

The previous command run failed yesterday but worked just now when I tried again. Looks so wired :(

But the `filepath` should wrap with double quotes or the command will fail if there is blank space in the filename.

**Comment by sarah11918** [comment]
Hi @CaptainOfPhB ! Would you please update your suggestion if you think all that's now needed are the double quotes? Then, we can get this merged!  :partying_face:

File: src/content/docs/en/recipes/modified-time.mdx [hunk 1]
```diff
@@ -43,7 +43,7 @@ Learn how to build a [remark plugin](https://github.com/remarkjs/remark) that ad
      export function remarkModifiedTime() {
        return function (tree, file) {
          const filepath = file.history[0];
-         const result = execSync(`git log -1 --pretty="format:%cI" ${filepath}`);
+         const result = execSync(`git --no-pager log -1 --pretty=%cI "${filepath}"`);
```
Comment (TheOtterlord) [inline src/content/docs/en/recipes/modified-time.mdx:46]:
```suggestion
         const result = execSync(`git log -1 --pretty="format:%cI" "${filepath}"`);
```

**Review by TheOtterlord — APPROVED** [review]

**Comment by TheOtterlord** [comment]
Thanks for the update! Let's get this merged! :partying_face:

**Pull request merged**